src/Symfony/Component/Runtime/RuntimeInterface.php

```diff
@@ -25,7 +25,7 @@ interface RuntimeInterface
      *
      * The callable itself should return an object that represents the application to pass to the getRunner() method.
      */
-    public function getResolver(callable $callable): ResolverInterface;
+    public function getResolver(callable $callable, \ReflectionFunction $reflector = null): ResolverInterface;
```

Comment:
`RuntimeInterface::getResolver()` now takes a second `\ReflectionFunction $reflector = null` argument. This allows injecting the reflector to use to analyse the `$callable`, providing both a small perf benefit when decorating, and allowing to decouple from the native implementation of `ReflectionFunction`.

Reply:
Could you show where you are using the second argument?

Reply:
I'm not, this is just applying the DI principle. Actually, I played with the idea a bit a few times already. Some applications I played with, but won't submit for now:
- read a PHP8 attribute on the $callable to select the ap-runtime
- decorate `GenericRuntime::getResolver()` in a child class
- build the `ReflectionFunction` instance out of static data (aka caching).

All those use cases need to create a ReflectionFunction instance before calling the method. It would be a shame if all decorates in the chain would have to create this again and again. Not to mention the last use case, which needs an actual injection point.